### PR TITLE
Loose ends in the "hook refactor"

### PR DESCRIPTION
**Pull request by richardolsson**
## Description
This PR ties up some loose ends in the hook refactor (#1555).

## Screenshots
None

## Changes
* Replaces some remnants of `react-query` in the tasks feature (using new mutation in `useTaskMutations`)
* Removes all other remnants of `react-query`
* Removes dead code related to models etc

## Notes to reviewer
Non

## Related issues
Related to #1542

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -71,7 +71,6 @@
     "react-final-form": "^6.5.9",
     "react-intl": "^6.1.0",
     "react-leaflet": "^4.2.1",
-    "react-query": "^3.39.2",
```
Comment (ziggabyte) [inline package.json:74]: 👏

File: src/pages/organize/[orgId]/projects/[campId]/calendar/tasks/[taskId]/assignees.tsx [hunk 2]
```diff
@@ -17,7 +13,9 @@ import TaskAssigneesList from 'features/tasks/components/TaskAssigneesList';
 import useAssignedTasks from 'features/tasks/hooks/useAssignedTasks';
 import { useMessages } from 'core/i18n';
 import { useNumericRouteParams } from 'core/hooks';
+import { useState } from 'react';
```
Comment (ziggabyte) [inline src/pages/organize/[orgId]/projects/[campId]/calendar/tasks/[taskId]/assignees.tsx:16]: this should be among the external imports 👼

File: src/pages/_app.tsx [hunk 3]
```diff
@@ -57,14 +55,6 @@ Router.events.on(
...
 );
 
-const queryClient = new QueryClient({
-  defaultOptions: {
-    queries: {
-      refetchOnWindowFocus: false,
-    },
-  },
-});
-
```
Comment (ziggabyte) [inline src/pages/_app.tsx:67]: Glad to see that this was such a simple thing, once all the parts that depended on it were gone!

**Review by ziggabyte — APPROVED** [review]
one tiiiiiiniest comment to change one line, otherwise it looks good to my eyes!

File: src/pages/organize/[orgId]/projects/[campId]/calendar/tasks/[taskId]/assignees.tsx (hunk 2 above)
Comment (richardolsson) [reply src/pages/organize/[orgId]/projects/[campId]/calendar/tasks/[taskId]/assignees.tsx:16]: In this file there is no distinction between internal and external. This has been the case in a couple of files, and I think it's a separate thing for us to figure out at some point, whether we want to stick to keeping them separate, and if so, enforce it somehow. So I'd rather not make any changes to this right now.

**Pull request merged**